Sources/CURLResponse.swift

```diff
@@ -300,6 +301,9 @@ public extension CURLResponse {
 	/// Get the response body decoded from JSON into a [String:Any] dictionary.
 	/// Invalid/non-JSON body data will result in an empty dictionary being returned.
 	public var bodyJSON: [String:Any] { do { return try bodyString.jsonDecode() as? [String:Any] ?? [:] } catch { return [:] } }
+	/// Get the response body decoded from JSON into a decodable structure
+	/// Invalid/non-JSON body data will result in nil
+	public func bodyJSON<T: Decodable>(_ type: T.Type) -> T? { do { return try JSONDecoder().decode(type, from: Data(bytes: bodyBytes)) } catch { return nil } }
```

Comment:
I think this should throw whatever the JSONDecoder throws. Instead of squelching the problem and giving no feedback.

Reply:
Adding this func is a good way to solve this one. Nice.

Reply:
The first thought was to throw but I changed my mind after reading `public var bodyJSON` ... surely I would like to throw!!! :-)